deps/build.jl

```diff
@@ -131,73 +120,142 @@ if binary == "system"
+                get(ENV, "JULIA_MPI_CLUSTER_WARN", "") != "n"
                 @warn """
                     You appear to be using MPI.jl with the default MPI binary on a cluster.
                     We recommend using the system-provided MPI, see the Configuration section of the MPI.jl docs.
                     """
             end
         end
     end
+
 elseif binary == "MPICH_jll"
+
     @info "using MPICH_jll"
+
+    using MPICH_jll
+    run(MPICH_jll.generate_compile_time_mpi_constants())
+
     deps = quote
         using MPICH_jll
-        include("consts_mpich.jl")
+        const libmpiconstants = MPICH_jll.libload_time_mpi_constants
         const _mpiexec = MPICH_jll.mpiexec
         const mpiexec_path = MPICH_jll.mpiexec_path
+
+        __init__deps() = nothing
+    end
+
+elseif binary == "MPItrampoline_jll"
+
+    @info "using MPItrampoline_jll"
+    using MPItrampoline_jll
+    @assert MPItrampoline_jll.is_available()
+    run(MPItrampoline_jll.generate_compile_time_mpi_constants())
+
+    deps = quote
+        @info "using MPItrampoline_jll"
+        using MPItrampoline_jll
+        @assert MPItrampoline_jll.is_available()
+        mpiexec_path = get(ENV, "MPITRAMPOLINE_MPIEXEC", nothing)
+        if mpiexec_path === nothing
+            if "MPITRAMPOLINE_LIB" ∉ keys(ENV)
+                @info "[MPI] Using built-in MPICH with MPItrampoline"
+                # In principle, MPItrampoline_jll.mpiwrapperexec should
+                # forward to MPItrampoline_jll.mpich_mpiexec. However,
+                # there is no mechanism to update the path to where the
+                # Julia artifact is installed, so we forward manually
+                # here.
+                const _mpiexec = MPItrampoline_jll.mpich_mpiexec
+                mpiexec_path = MPItrampoline_jll.mpich_mpiexec_path
+            else
+                const _mpiexec = MPItrampoline_jll.mpiwrapperexec
+                mpiexec_path = MPItrampoline_jll.mpiwrapperexec_path
+            end
+        else
+            const mpiexec_cmd = Cmd([mpiexec_path])
+            _mpiexec(fn) = fn(mpiexec_cmd)
+        end
+        const libmpiconstants = MPItrampoline_jll.libload_time_mpi_constants_path
+    
         __init__deps() = nothing
     end
+
 elseif binary == "OpenMPI_jll"
+
     @info "using OpenMPI_jll"
+
+    using OpenMPI_jll
+    run(OpenMPI_jll.generate_compile_time_mpi_constants())
+
     deps = quote
         using OpenMPI_jll
-        include("consts_openmpi.jl")
         const _mpiexec = OpenMPI_jll.mpiexec
         const mpiexec_path = OpenMPI_jll.mpiexec_path
 
         function __init__deps()
             # Required for OpenMPI relocateable binaries
-            # TODO: this should be done in OpenMPI_jll package
+            # TODO: this should be done in the OpenMPI_jll package
             # https://github.com/JuliaPackaging/Yggdrasil/issues/390
             ENV["OPAL_PREFIX"] = OpenMPI_jll.artifact_dir
```

Comment:
Done in https://github.com/JuliaPackaging/Yggdrasil/pull/3967

Reply:
Done.